pkg/cluster/workloadidentityresources.go

```diff
@@ -124,3 +126,15 @@ func (m *manager) generateAuthenticationConfig() (*configv1.Authentication, erro
 		},
 	}, nil
 }
+
+func (m *manager) getPlatformWorkloadIdentityFedertedCredName(identity api.PlatformWorkloadIdentity) (string, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return "", fmt.Errorf("getPlatformWorkloadIdentityFedertedCredName called for a CSP cluster")
+	}
+
+	identityResourceId, err := azure.ParseResourceID(identity.ResourceID)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s-%s", m.doc.ID, identityResourceId.ResourceName), nil
```

Comment:
Any justification for this naming convention?

Want to make sure it fits within the requirements for Azure resource naming. For example, how long is `doc.ID`?

Reply:
https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation-considerations#errors gives an idea about the naming.
Also when creating federated at portal:-
`The name of a federated identity credentials must be within 3 and 120 characters, only contains letters (A-Z, a-z), numbers, hyphens and dashes and must start with a number or letter.`

Reply:
Given that we are constructing a name with `m.doc.ID` and `identityResourceId.ResourceName`. I think we could assume all those conditions are satisfied and no need to add logic to protect against that?